src/action-ros-ci.ts

```diff
@@ -111,6 +111,42 @@ export async function execBashCommand(
 	});
 }
 
+//list of valid ROS distributions
+const validROS1Distros: string[] = ["kinetic", "lunar", "melodic", "noetic"];
+const validROS2Distros: string[] = ["dashing", "eloquent", "foxy"];
+
+//Determine whether all inputs name supported ROS distributions.
+export function validateDistro(obj: { t1; t2; cmd }): boolean {
+	if (!obj.t1 && !obj.t2) {
+		obj.cmd +=
+			"Neither `target_ros1_distro` or `target_ros2_distro` inputs were set, at least one is required.";
+
+		return false;
+	}
+	if (obj.t1) {
+		if (validROS1Distros.indexOf(obj.t1) <= -1) {
+			obj.cmd += `Input ${obj.t1} was not a valid ROS 1 distribution for \`target_ros1_distro\`.' Valid values: ${validROS1Distros}`;
+
+			return false;
+		}
+		if (process.platform == "linux") {
+			obj.cmd += `mkdir -p /opt/ros/${obj.t1} && touch /opt/ros/${obj.t1}/setup.sh} && source /opt/ros/${obj.t1}/setup.sh && `;
+		}
+	}
+	if (obj.t2) {
+		if (validROS2Distros.indexOf(obj.t2) <= -1) {
+			obj.cmd += `Input ${obj.t2} was not a valid ROS 2 distribution for \`target_ros2_distro\`. Valid values: ${validROS2Distros}`;
+
+			return false;
+		}
+		if (process.platform == "linux") {
+			obj.cmd += `mkdir -p /opt/ros/${obj.t2} && touch /opt/ros/${obj.t2}/setup.sh} && source /opt/ros/${obj.t2}/setup.sh && `;
```

Comment:
> `setup.sh}`

I assume this is a typo?

Reply:
Yes, definitely - why did this work in the tests...?

Reply:
This is fixed and the logic is clarified in https://github.com/ros-tooling/action-ros-ci/pull/289/files

Reply:
And I now see why this wasn't triggered by the tests, i've updated #289 to have a test that would have failed here. Normally I'd roll this change back, but we haven't released a new version yet so I think we should just move forward with the fix in the follow-up PR since it's already open.